packages/headless/src/ssr-next/commerce/engine/commerce-engine.ssr.ts

```diff
@@ -16,18 +15,44 @@ import type {CommerceEngineDefinition} from '../types/engine.js';
 /**
  * Initializes a Commerce engine definition in SSR with given controllers definitions and commerce engine config.
  * @param options - The commerce engine definition
- * @returns Three utility functions to fetch the initial state of the engine in SSR, hydrate the state in CSR,
- *  and a build function that can be used for edge cases requiring more control.
+ * @returns Engine definitions for different solution types that support per-call navigator context.
  *
  * @remarks
  * You can use the {@link InferStaticState} and {@link InferHydratedState} utility types with the returned engine definitions
  * to infer the types of static and hydrated state for your controllers.
  *
  * @example
  * ```ts
- * const engineDefinition = defineCommerceEngine(engineConfig);
- * type SearchStaticState = InferStaticState<typeof engineDefinition>;
- * type SearchHydratedState = InferHydratedState<typeof engineDefinition>;
+ * const { listingEngineDefinition } = defineCommerceEngine(engineConfig);
+ *
+ * // Pass navigator context directly to fetchStaticState
+ * const staticState = await listingEngineDefinition.fetchStaticState({
+ *   navigatorContext: {
+ *     forwardedFor: req.ip,
+ *     referrer: req.headers.referer,
+ *     userAgent: req.headers['user-agent'],
+ *     location: req.url,
+ *     clientId: 'unique-session-id'
+ *   }
+ * });
+ *
+ * // Framework examples:
+ * // Express.js
+ * app.get('/api/listing', async (req, res) => {
+ *   const state = await listingEngineDefinition.fetchStaticState({
+ *     navigatorContext: {
+ *       forwardedFor: req.ip,
+ *       referrer: req.get('Referer'),
```

Comment:
Same thing here.
The referrer might be required in the context (in the future).
https://coveord.atlassian.net/wiki/spaces/~712020b50e23f6c42e4fb78119f2144c0e859f/pages/5397577741/EPIC+RESULTS+Validate+that+query+parameters+are+passed+by+Headless+and+handled+by+CAPI